internal/juju/offers.go

```diff
@@ -195,26 +195,30 @@ func (c offersClient) DestroyOffer(input *DestroyOfferInput) error {
 		return err
 	}
 
-	forceDestroy := false
-	//This code loops until it detects 0 connections in the offer or 3 minutes elapses
-	if len(offer.Connections) > 0 {
-		end := time.Now().Add(5 * time.Minute)
-		for ok := true; ok; ok = len(offer.Connections) > 0 {
-			//if we have been failing to destroy offer for 5 minutes then force destroy
-			//TODO: investigate cleaner solution (acceptance tests fail even if timeout set to 20m)
-			if time.Now().After(end) {
-				forceDestroy = true
-				break
-			}
-			time.Sleep(10 * time.Second)
-			offer, err = client.ApplicationOffer(input.OfferURL)
-			if err != nil {
-				return err
+	checkConnections := func() error {
+		//This code loops until it detects 0 connections in the offer or 3 minutes elapses
+		if len(offer.Connections) > 0 {
+			end := time.Now().Add(5 * time.Minute)
+			for ok := true; ok; ok = len(offer.Connections) > 0 {
+				//if we have been failing to destroy offer for 5 minutes then fail on destroy
+				if time.Now().After(end) {
+					return fmt.Errorf("offer %q has remaining connections", input.OfferURL)
```

Comment:
possibly add to the log the remaining connections, so a final user can debug it/file a more precise bug report

Reply:
@SimoneDutto good idea, I suggest different output though. Not all users will understand connections. 

RelationID would be ideal, but the provider doesn't deal in RelationIDs like the juju cli does.

Let's try offer.SourceModelUID and offer.Endpoint in the message. It should have the `application:relation` in the string. That should be enough info to find the relation causing the issue.